### PR TITLE
SNOW-658388: Check session parameter is None

### PR DESCRIPTION
**Pull request by sfc-gh-aling**
Please answer these questions before submitting your pull requests. Thanks!

1. What GitHub issue is this PR addressing? Make sure that there is an accompanying issue to your PR.

   Fixes SNOW-658388

`_session_parameter` could be `None` is stored proc, we should check and default to `True` if it's None.

2. Fill out the following pre-review checklist:

   - [x] I am adding a new automated test(s) to verify correctness of my new code
   - [ ] I am adding new logging messages
   - [ ] I am adding a new telemetry message
   - [ ] I am adding new credentials
   - [ ] I am adding a new dependency

3. Please describe how your code solves the related issue.

   Please write a short description of how your code change solves the related issue.

### CODE REVIEW

File: src/snowflake/snowpark/session.py [hunk 1]
```diff
@@ -259,10 +259,14 @@ def __init__(self, conn: ServerConnection) -> None:
...
+        self._use_scoped_temp_objects = bool(
             _use_scoped_temp_objects
-            and conn._conn._session_parameters.get(
-                _PYTHON_SNOWPARK_USE_SCOPED_TEMP_OBJECTS_STRING, True
+            and (
+                conn._conn._session_parameters.get(
+                    _PYTHON_SNOWPARK_USE_SCOPED_TEMP_OBJECTS_STRING, True
+                )
+                if conn._conn._session_parameters
+                else True
```
Comment (sfc-gh-sfan) [inline src/snowflake/snowpark/session.py:269]: We don't have to do it in this PR, but feels like if we have more similar usage, we should create a helper function :)

**Review by sfc-gh-sfan — APPROVED** [review]

**Comment by sfc-gh-sfan** [comment]
Can you also run a stored proc regression test with this change?

File: src/snowflake/snowpark/session.py (hunk 1 above)
Comment (sfc-gh-aling) [reply src/snowflake/snowpark/session.py:269]: good idea, I could make a comment for this

**Review by sfc-gh-jdu — APPROVED** [review]

**Comment by sfc-gh-sfan** [comment]
BUWA: https://buwa.c1.us-west-2.aws-dev.app.snowflake.com/builds/precommit/pc-jenkins.int.snowflakecomputing.com/53717
Precommit: https://jenkins.int.snowflakecomputing.com/job/PythonStoredProcBuildPrecommitTest/79/

AWS tests passed. The other two test suites (azure, local) failed due to unrelated reasons

```
2022-09-09 16:15:58  [INFO] Connecting to FDB
2022-09-09 16:15:58  Error: The directory does not exist.
2022-09-09 16:15:58  [INFO] Done setting up metadata FoundationDB database 
```

It should provide enough confidence that this would fix the stored proc error exposed this afternoon.

**Pull request merged**